uktrade/data-hub-frontend
# Adding Activity Stream activities to a Contact

## Description of change

We want to show all activities from Activity Stream for a Contact under the Contact Interactions tab, so that we can surface Aventri data for users and advisers can have a full picture of the contact's activity.

This story began with wanting to just pull in Aventri data, but we found with a query to Activity Stream, and by re-using much of the Activity Feed architecture, we could pull in other external and Data Hub data sources.

NOTE: We'll be making iterative releases and this is the very first chunk, which we're hiding behind a user feature flag. It's not intended for production users, but will allow us to do some testing and see how it looks with prod data.

## Test instructions

There is currently only Aventri data for one contact, John Doe, and only on the dev environment.
If you are running locally, you'll need to:

1. point at dev environment
2. add the `user-contact-activities` user feature flag to your user (you can do this in the Django admin -> Adviser > Other)
3. Visit John Doe: http://localhost:3000/contacts/68be55fc-544d-4d9c-a264-977b917f1fd3/interactions and see the data coming through 

## Screenshots
### Before

![Screenshot 2022-03-25 at 16 02 50](https://user-images.githubusercontent.com/16647486/160157587-011083e6-da5d-482f-8a55-34471f38b951.png)


### After
(with feature flag active on user, should look the same to users without flag)

![Screenshot 2022-03-25 at 16 03 11](https://user-images.githubusercontent.com/16647486/160157615-1697e39e-8b41-4242-9dfb-0b7905a85490.png)



## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
